src/backend/metal/src/device.rs

```diff
@@ -1079,14 +1079,6 @@ impl hal::Device<Backend> for Device {
         let pipeline_layout = &pipeline_desc.layout;
         let pass_descriptor = &pipeline_desc.subpass;
 
-        let attribute_buffer_index = pipeline_layout.attribute_buffer_index();
-        if attribute_buffer_index + pipeline_desc.vertex_buffers.len() as ResourceIndex > self.private_caps.max_buffers_per_stage {
```

Comment:
Will something else catch this case now?

Reply:
there is another check in the loop below that catches this more robustly